File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    var mp = new Map();
    var j=0;
    var ans=[];
    nums.forEach((i)=>{
        if(mp.get(target-i)){
            ans.push(j);
            ans.push(mp.get(target-i)-1);
            // console.log(ans);
        }
        mp.set(i,j+1);
        j++;
    })
    return ans;
};

File: 0001-two-sum/0001-two-sum.go
func twoSum(nums []int, target int) []int {
    twoS := make(map[int]int);
    for i:=0; i<len(nums);i++{
        if j, prs := twoS[nums[i]]; prs {
            return  []int{j, i}
        }
        twoS[target-nums[i]] = i
    }
    return nil
}